### PR TITLE
EXADS Bid Adapter: initial release

### PR DESCRIPTION
**Pull request by giuseppe-exads**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Other information

doc - https://github.com/prebid/prebid.github.io/pull/5237

### CODE REVIEW

File: modules/exadsBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+  utils.logInfo('Page -> ', envParams.page);
+  utils.logInfo('Lang -> ', envParams.lang);
+  utils.logInfo('OS -> ', envParams.osName);
+  utils.logInfo('User Agent -> ', envParams.userAgent);
+  utils.logInfo('Primary Language -> ', envParams.language);
+
+  return envParams;
+}
+
+export const imps = new Map();
```
Comment (muuki88) [inline modules/exadsBidAdapter.js:358]: Why exporting a global map?

File: modules/exadsBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+  if (hasGdprConsent) {
+    return bidderRequest.gdprConsent;
+  }
+
+  return null;
+}
+
+export const spec = {
+  aliases: ['exads'], // short code
+  supportedMediaTypes: [BANNER, NATIVE, VIDEO],
```
Comment (muuki88) [inline modules/exadsBidAdapter.js:436]: You are missing your `GVL_ID` here ( 1084 )

File: modules/exadsBidAdapter.js (hunk 1 above)
Comment (giuseppe-exads) [reply modules/exadsBidAdapter.js:358]: I'm exporting it in order to make it testable. I'm using it into unit tests. An alternative is to make it as a property of the "spec" object.

File: modules/exadsBidAdapter.js (hunk 2 above)
Comment (giuseppe-exads) [reply modules/exadsBidAdapter.js:438]: Yes, you are good. Just put the GVL_ID.

**Comment by Rothalack** [comment]
Hey @giuseppe-exads , I'm trying to test and I have been unable to get any valid bid to go through. It's always either invalid or errors out. Do you have a set of params that would work for a test?

**Comment by giuseppe-exads** [comment]
> Hey @giuseppe-exads , I'm trying to test and I have been unable to get any valid bid to go through. It's always either invalid or errors out. Do you have a set of params that would work for a test?

Hey @Rothalack,

I've configured one test for you on our adservers. It is related to a banner 300x250. You can interact with it using our prebidJS adapter, using the next model and param.

Attention to some parameters! For some params you need to specify your custom values. Read the comments near to some params for more information.

I've tested it on my side and it works.

 Let me know if you need other information and /or other tests. Thanks so much.

```
{
    "code": "postbid_iframe",  // specify your value based on your web site
    "mediaTypes": {
        "banner": {
            "sizes": [
                [
                    300,
                    250
                ]
            ]
        }
    },
    "bids": [
        {
            "bidder": "exadsadserver",
            "params": {
                "zoneId": 390,
                "fid": "885643a525fc065497d80df32141dfcce4b9447a",
                "partner": "ortb_2_4",
                "siteId": "12345",
                "siteName": "www.google.com",
                "userIp": "37.228.226.18", // your user ip
                "userId": "2177189013422152",
                "country": "IRL", // your country code
                "impressionId": "8694154782644030", // you can leave it or specify one custom id
                "keywords": "Category 1",
                "bidfloor": 1.1e-7,
                "bidfloorcur": "EUR",
                "bcat": [
                    "IAB25",
                    "IAB7-39",
                    "IAB8-18",
                    "IAB8-5",
                    "IAB9-9"
                ],
                "badv": [
                    "first.com",
                    "second.com"
                ],
                "mimes": [
                    "image/jpg"
                ],
                "dsa": {
                    "dsarequired": 3,
                    "pubrender": 0,
                    "datatopub": 2
                },
                "imageOutput": "html",
                "endpoint": "https://rtb.adnflow.com/rtb.php"
            }
        }
    ]
}
```

**Review by Rothalack — APPROVED** [review]
Sorry for the delay, I thought I had already approved and merged this. This is working in testing for me, thanks for the testing set up!

**Comment by giuseppe-exads** [comment]
Thanks @Rothalack!

**Comment by giuseppe-exads** [comment]
@tfoliveira

**Pull request merged**